src/cli.rs

```diff
@@ -942,10 +942,160 @@ pub async fn main<H: Hooks>() -> crate::Result<()> {
     Ok(())
 }
 
+// Define route node structure with enhanced methods
+#[derive(Default)]
+struct RouteNode {
+    children: BTreeMap<String, RouteNode>,
+    endpoints: Vec<(String, String)>,
+}
+
+impl RouteNode {
+    fn is_leaf(&self) -> bool {
+        self.endpoints.len() == 1 && self.children.is_empty()
+    }
+
+    fn is_collapsible(&self) -> bool {
+        self.endpoints.is_empty()
+            && self.children.len() == 1
+            && self.children.values().next().unwrap().is_leaf()
+    }
+
+    fn method(&self) -> &str {
+        &self.endpoints[0].0
```

Comment:
Accessing `&self.endpoints[0]` can panic if the `endpoints` list is empty.  
Consider using `self.endpoints.get(0)` instead for safe access.